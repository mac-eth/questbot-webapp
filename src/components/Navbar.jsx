import { useState } from "react";

import { close, logo, menu } from "../assets";
import { navLinks } from "../constants";
import styles from "../style";
import LoginButton from "./LoginButton";

const Navbar = () => {
  const [active, setActive] = useState("Home");
  const [toggle, setToggle] = useState(false);

  return (
    <nav className="w-full flex py-6 justify-between items-center ">
      <img src={logo} alt="QuestBot" className="w-[134px] h-[32px]" />

      <ul className="list-none sm:flex hidden justify-end items-center flex-1">
        {navLinks.map((nav, index) => (
          <li
            key={nav.id}
            className={`font-poppins font-normal cursor-pointer text-[16px] ${
              active === nav.title ? "text-white" : "text-dimWhite"
            } ${index === navLinks.length - 1 ? "mr-0" : "mr-10"}`}
            onClick={() => setActive(nav.title)}
          >
            <a href={`#${nav.id}`}>{nav.title}</a>
          </li>
        ))}
      </ul>
      <LoginButton />

      <div className="sm:hidden flex flex-1 justify-end items-center">
        <img
          src={toggle ? close : menu}
          alt="menu"
          className="w-[28px] h-[28px] object-contain"
          onClick={() => setToggle(!toggle)}
        />

        <div
          className={`${
            !toggle ? "hidden" : "flex"
          } p-6 bg-black-gradient absolute top-20 right-0 mx-4 my-2 min-w-[140px] rounded-xl sidebar`}
        >
          <ul className="list-none flex justify-end items-start flex-1 flex-col">
            {navLinks.map((nav, index) => (
              <li
                key={nav.id}
                className={`font-poppins font-medium cursor-pointer text-[16px] ${
                  active === nav.title ? "text-white" : "text-dimWhite"
                } ${index === navLinks.length - 1 ? "mb-0" : "mb-4"}`}
                onClick={() => setActive(nav.title)}
              >
                <a href={`#${nav.id}`}>{nav.title}</a>
              </li>
            ))}
          </ul>
          <div
            className={`${styles.flexCenter}  ml-10 w-[80px] h-[40px] rounded-[10px] bg-purple-gradient glow-on-hover-small  p-[2px] cursor-pointer`}
          >
            <div
              className={`${styles.flexCenter} flex-col bg-black w-[100%] h-[100%] rounded-[10px]`}
            >
              <div className={`${styles.flexStart} flex-row`}>
                <div className="font-poppins font-medium text-[16px]">
                  <div className="text-dimWhite hover:text-white active:text-white">
                    <a href="#login">Login</a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </nav>
  );
};

export default Navbar;
