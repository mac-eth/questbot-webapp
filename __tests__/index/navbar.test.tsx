import { render, screen } from "@testing-library/react";
import Renderer from "react-test-renderer";
import Navbar from "../../src/components/layout/Navbar";
import "@testing-library/jest-dom";
import { SessionProvider } from "next-auth/react";

describe("Navbar Component", () => {
  test.skip("Renders the Navbar Logo", () => {
    render(
      <SessionProvider>
        <Navbar />
      </SessionProvider>
    );
    const logo = screen.getByRole("img", {
      name: "questbot_logo",
    });
    expect(logo).toBeInTheDocument();
  });
  it("Renders the entire Navbar", () => {
    const renderer = Renderer.create(
      <SessionProvider>
        <Navbar />
      </SessionProvider>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
