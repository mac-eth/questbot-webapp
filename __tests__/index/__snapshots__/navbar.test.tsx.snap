// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component Renders the entire Navbar 1`] = `
<div
  style={
    {
      "transform": "translateY(-100px) translateZ(0)",
    }
  }
>
  <nav
    className="relative w-full flex py-6 justify-between items-center z-[10] "
  >
    <span
      style={
        {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
          style={
            {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        alt="QuestBot"
        className="w-[134px] h-[32px]"
        data-nimg="intrinsic"
        decoding="async"
        onError={[Function]}
        onLoad={[Function]}
        role="QuestBot"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="QuestBot"
          className="w-[134px] h-[32px]"
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          role="QuestBot"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style={
            {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
    <ul
      className="list-none sm:flex hidden justify-end items-center flex-1"
    >
      <li
        className="font-poppins font-normal cursor-pointer text-[16px] hover:underline text-dimWhite mr-10"
        onClick={[Function]}
      >
        <a
          href="#"
        >
          Home
        </a>
      </li>
      <li
        className="font-poppins font-normal cursor-pointer text-[16px] hover:underline text-dimWhite mr-10"
        onClick={[Function]}
      >
        <a
          href="#docs"
        >
          Docs
        </a>
      </li>
      <li
        className="font-poppins font-normal cursor-pointer text-[16px] hover:underline text-dimWhite mr-0"
        onClick={[Function]}
      >
        <a
          href="#dashboard"
        >
          Dashboard
        </a>
      </li>
    </ul>
    <div
      className="flex justify-center items-center  ml-10 w-[80px] h-[40px] rounded-[10px] bg-purple-gradient glow-on-hover-small  p-[2px] cursor-pointer"
    >
      <div
        className="flex justify-center items-center flex-col bg-black w-[100%] h-[100%] rounded-[10px]"
      >
        <div
          className="flex justify-center items-start flex-row"
        >
          <div
            className="font-poppins font-medium text-[16px]"
          >
            <div
              className="text-dimWhite hover:text-white active:text-white"
            >
              <button
                onClick={[Function]}
              >
                Login
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sm:hidden flex flex-1 justify-end items-center"
    >
      <span
        style={
          {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
            style={
              {
                "background": "none",
                "border": 0,
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          />
        </span>
        <img
          alt="menu"
          className="w-[28px] h-[28px] object-contain"
          data-nimg="intrinsic"
          decoding="async"
          onClick={[Function]}
          onError={[Function]}
          onLoad={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="menu"
            className="w-[28px] h-[28px] object-contain"
            data-nimg="intrinsic"
            decoding="async"
            loading="lazy"
            onClick={[Function]}
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style={
              {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
      <div
        className=" p-6 bg-black-gradient absolute top-20 right-0 mx-4 my-2 min-w-[140px] rounded-xl sidebar"
      >
        <ul
          className="list-none flex justify-end items-start flex-1 flex-col"
        >
          <li
            className="font-poppins font-medium cursor-pointer text-[16px] hover:underline text-dimWhite mb-4"
            onClick={[Function]}
          >
            <a
              href="#"
            >
              Home
            </a>
          </li>
          <li
            className="font-poppins font-medium cursor-pointer text-[16px] hover:underline text-dimWhite mb-4"
            onClick={[Function]}
          >
            <a
              href="#docs"
            >
              Docs
            </a>
          </li>
          <li
            className="font-poppins font-medium cursor-pointer text-[16px] hover:underline text-dimWhite mb-0"
            onClick={[Function]}
          >
            <a
              href="#dashboard"
            >
              Dashboard
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
